server/coordinator/procedure/batch_procedure.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package procedure
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/CeresDB/ceresmeta/pkg/log"
+	"github.com/CeresDB/ceresmeta/server/storage"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+// BatchProcedure is a proxy procedure contains a batch of procedures with same type and version.
+// It is used to support concurrent execution of a batch of procedures with the same type and version.
+type BatchProcedure struct {
+	id                 uint64
+	typ                Typ
+	batch              []Procedure
+	relatedVersionInfo RelatedVersionInfo
+
+	// Protect the state.
+	lock  sync.RWMutex
+	state State
+}
+
+func NewBatchProcedure(id uint64, batch []Procedure, typ Typ) (Procedure, error) {
```

Comment:
I thought about it for a while, and there seems to be no special benefit in doing so, and other procedures are also doing the same. If I want to modify it, it will involve many places, so I decided not to make adjustments for the time being.